source/agora/node/FullNode.d

```diff
@@ -491,6 +491,32 @@ public class FullNode : API
         if (this.pool.hasTransactionHash(tx_hash))
             return;
 
+        // all inputs have to be known in order to calculate the fee and
+        // compare it to the fee threshold
+        UTXO utxo;
+        foreach (const ref input; tx.inputs)
+            if (!utxo_set.peekUTXO(input.utxo, utxo))
+                return;
+
+        // only consider a double spend transaction, if its fee is
+        // considerably higher than the current highes fee
+        auto fee_threshold = ledger.getDoubleSpentHighestFee(tx);
+
+        // if the fee_threshold is null, it means there won't be any double
+        // spend transactions, after this transaction is added to the pool
+        if (!fee_threshold.isNull())
+        {
+            fee_threshold.div(100);
```

Comment:
This ignore the remainder, which means it might not respect the percentage

Reply:
`percentage()` + `add()` is probably the right way to go.

Reply:
@omerfirmak not sure I understand your 'percentage() + add()' solution, let's say the amount is 199 and percentage is 20%

are you suggesting 199 + (199/100)*20 = 219? If we had real numbers then the result should be 238.8.

if we could just turn out the overflow check and would calculate 199*120/100 that would be 238 which is very close to 238.8

Reply:
in general if we do something like Y/X then the biggest mistake of integer calculation will be (X-1)/X and we amplifie it by 20 when we multiply. It would be better to multiply first and then divide where our error is (X-1/X)

Reply:
@ferencdg , `Amount.percentage()`will first multiply then div by 100. So it should have minimal error.

Reply:
@ferencdg : To clarify: with an amount of 199 and 20%, we should only accept a fee `>= 239` if we can't represent `238.8`. Even if the result of the calculation was `238.00001` it should only allow `>= 239`.

Reply:
> @ferencdg , `Amount.percentage()`will first multiply then div by 100. So it should have minimal error.

that can overflow though...

Reply:
With our current `Amount.MaxUnitSupply` it can't overflow.

Reply:
> With our current `Amount.MaxUnitSupply` it can't overflow.

what if an attacker sends a fee of MaxUnitSupply - 1? At this point no checks are mode on whether the transaction is valid.

Reply:
In theory we can have a transaction with a fee > Maxsupply/20, and it will overflow ;) while the
199 + (99*20/100) + (199/100)*20 solution not

Reply:
> @ferencdg : To clarify: with an amount of 199 and 20%, we should only accept a fee `>= 239` if we can't represent `238.8`. Even if the result of the calculation was `238.00001` it should only allow `>= 239`.

so do you know how to estimate the error, so I can correct it? if we use the 199 + (99*20/100) + (199/100)*20 formula... Btw, do we really want to be this precise? I just randomly choose 20%....

Reply:
I was solely talking about the call to `percentage()`. I don't think it will overflow, ever. But after you add the `percentage()` result to the fee, it might overflow yes. At that point we can either accept or reject that TX.